src/Strategy/Elixir/MixTree.hs

```diff
@@ -350,42 +339,31 @@ toDependencyVersion dt = case parse parseConstraintExpr "" dt of
 parseVerConstraint :: Parser VerConstraint
 parseVerConstraint = do
   operator <- whitespaceOrTab *> parseConstraintOperator <* whitespaceOrTab
-  versionText <- findVersionText <* whitespaceOrTab
+  versionText <-  findVersionText <* whitespaceOrTab
   case operator of
-    Equal -> pure $ CEq versionText
-    NotEqual -> pure $ CNot versionText
-    GreaterThanOrEqual -> pure $ CGreaterOrEq versionText
-    GreaterThan -> pure $ CGreater versionText
-    LessThanOrEqual -> pure $ CLessOrEq versionText
-    LessThan -> pure $ CLess versionText
-    Compatible -> pure $ CCompatible versionText
-    WildcardAny -> pure $ CEq "*"
+    Just "==" -> return $ CEq versionText
+    Just "=" -> return $ CEq versionText
+    Just "!=" -> return $ CNot versionText
+    Just ">=" -> return $ CGreaterOrEq versionText
+    Just ">" -> return $ CGreater versionText
+    Just "<=" -> return $ CLessOrEq versionText
+    Just "<" -> return $ CLess versionText
+    Just "~>" -> return $ CCompatible versionText
+    Just "*" -> return $ CEq "*"
+    Just _ -> fail ("Could not recognize mix constraint operator:")
+    Nothing -> return $ CEq versionText
   where
-
     whitespaceOrTab :: Parser Text
     whitespaceOrTab = takeWhileP (Just "whitespaceOrTab") (\c -> c == ' ' || c == '\t')
 
+    parseConstraintOperator :: Parser (Maybe Text)
+    parseConstraintOperator = optional (asum (map symbol operatorList))
```

Comment:
[nit]: Use `$` and `.` instead of brackets in easy cases like this.
```suggestion
    parseConstraintOperator = optional . asum $ map symbol operatorList
```